platform/marvell-arm64/platform.conf

```diff
@@ -4,30 +4,134 @@
-fdt_fname="/boot/armada-7020-comexpress.dtb"
+kernel_addr=0x1100000
+kernel_version=5.10.0-18-2-arm64
+kernel_fname="/boot/vmlinuz-$kernel_version"
+initrd_fname="/boot/initrd.img-$kernel_version"
 fit_fname="/boot/sonic_arm64.fit"
+demo_volume_label=SONiC-OS
 
 # global mount defines
-#demo_dev=/dev/sda1
 demo_mnt=/tmp
-#mtd_dev=/dev/$(cat /proc/mtd | grep "ENV" | grep -o "mtd[0-9]")
-FW_ENV_DEFAULT='/dev/mtd1 0x0 0x10000 0x100000'
-UBOOT_FW_DEFAULT=1
+
+if [ "$install_env" = "onie" ]; then
+    MACH_FILE="/etc/machine.conf"
+else
+    MACH_FILE="/host/machine.conf"
+fi
+
+PLATFORM=`sed -n 's/^onie_platform=\(.*\)/\1/p' $MACH_FILE`
+echo "Intalling SONiC from $install_env on Platform $PLATFORM"
+
+case $PLATFORM in
+    arm64-nokia_ixs7215_52xb-r0) PLATFORM_AC5X=1 ;;
+    *) PLATFORM_AC5X=0;;
+esac
+
+if [ $PLATFORM_AC5X -eq 1 ]; then
+    fdt_addr=0x201000000
+    fit_addr=0x210000000
+    initrd_addr=0x206000000
+
+    fdt_fname="/boot/ac5x.dtb"
+
+    FW_ENV_DEFAULT='/dev/mtd0 0x400000 0x10000 0x10000'
+    demo_part=2
+    fit_conf_name="#conf_ac5x"
+else
+    fdt_addr=0x1000000
+    fit_addr=0x8000000
+    initrd_addr=0x2000000
+
+    fdt_fname="/boot/armada-7020-comexpress.dtb"
+
+    FW_ENV_DEFAULT='/dev/mtd1 0x0 0x10000 0x100000'
+    demo_part=1
+fi
 
 # Skip VID Header in UBIFS
 LINUX_MISC_CMD='apparmor=1 security=apparmor usbcore.autosuspend=-1'
 
+#Get block device
+#Default block device is eMMC, if not look for usb storage
+get_install_device()
+{
+    if [ $PLATFORM_AC5X -eq 1 ]; then
+        mmc_bus="mmc0:0001"
+    else
+        mmc_bus="mmc0:aaaa"
+    fi
+    for i in 0 1 2 ; do
+        if $(ls -l /sys/block/mmcblk$i/device 2>/dev/null | grep -q "$mmc_bus") ; then
+            echo "/dev/mmcblk$i"
+            blk_dev=/dev/mmcblk$i
+            echo "Selected mmc $blk_dev"
+            return 0
+        fi
+    done
+
+    echo "ERROR storage not found"
+    return 1
+}
+
+get_install_device
+if [ $? -ne 0 ]; then
+    echo "Error: Unable to detect $blk_dev $demo_dev"
+    exit 1
+fi
+
+demo_dev=${blk_dev}p${demo_part}
+
+remove_dev_partitions() {
+    echo "Remove all existing partitions starting partnum: ${demo_part} from ${blk_dev}"
+    local dev_to_install=${blk_dev}p
+    for p in $(seq ${demo_part} 9) ; do
+        if [[ -e ${dev_to_install}${p} ]]; then
+            echo "Removing partition ${dev_to_install}${p}"
+            sgdisk -d ${p} ${blk_dev} || true
+        fi
+    done
+    partprobe ${blk_dev}
+}
+
+create_demo_partition() {
+    # SD CARD
+    remove_dev_partitions
+
+    # Create sonic partition
+    sgdisk --new ${demo_part}:: \
+        --change-name=${demo_part}:${demo_volume_label} \
+        --typecode=${demo_part}:8300 -p ${blk_dev}
+
+    partprobe
+}
+
+create_partition() {
+    get_install_device
+    if [ $? -ne 0 ]; then
+        echo "Error: Unable to detect $blk_dev $demo_dev"
+        exit 1
+    fi
+
+    # Platform specific partition 
+    create_demo_partition
+}
+
+mount_partition() {
```

Comment:
i couldn't find where create_partition and mount_partition called in this script?

Reply:
These functions are not called in this script.

They are called from installer/install.sh (line 165) which sources this platform.conf file.
```
. ./default_platform.conf

if [ -r ./platform.conf ]; then
    . ./platform.conf
fi

image_dir="image-$image_version"

if [ "$install_env" = "onie" ]; then
    # Create/format the flash
    create_partition
    mount_partition
```